### PR TITLE
feat: Ownable2Step coinbase

### LINKED ISSUE #1488: feat: Upgrade OZ and use Ownable2Step for Coinbase
### Describe the Feature Request

`Ownable2Step` is safe as it allows to make sure that the new owner will be able to withdraw funds (because it allows to make sure that the new owner is able to trigger a tx to the contract).

### What

- [ ] Update OZ submodule to the latest audited version (V5.0.2 at the time of writing this issue)
- [ ] Make required changes to existing solidity files so that it builds and works (beware of the updated remappings in the OZ lib itself)
- [ ] Use `Ownable2Step` for `Coinbase.sol` and update end to end tests

### PR DESCRIPTION
**Pull request by obatirou**
Time spent on this PR:

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #1488

## What is the new behavior?

Coinbase use Ownable2Step from OZ with version 5.0.2

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot/1494)

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -13,3 +13,6 @@
 [submodule "solidity_contracts/lib/base64-sol"]
 	path = solidity_contracts/lib/base64-sol
 	url = https://github.com/Brechtpd/base64
+[submodule "solidity_contracts/lib/openzeppelin-contracts"]
+	path = solidity_contracts/lib/openzeppelin-contracts
+	url = https://github.com/OpenZeppelin/openzeppelin-contracts
```
Comment (ClementWalter) [inline .gitmodules:18]: already un submodules

**Review by ClementWalter — CHANGES_REQUESTED** [review]
just the openzeppeline contracts are already in the submodules

**Comment by obatirou** [comment]
> just the openzeppeline contracts are already in the submodules

Yes there are used in the tests for uniswapv3. The version of the library used for the uniswapv3 is 3.4 which will bring breaking changes leading to code modifications to uniswapv3
My suggestion here is to keep both. 
It is also possible to delete the uniswapv3 test and code as it is skipped

File: .gitmodules (hunk 1 above)
Comment (obatirou) [reply .gitmodules:18]: Yes there are used in the tests for uniswapv3. The version of the library used for the uniswapv3 is 3.4 which if upgraded to 5.0 will bring breaking changes leading to code modifications to uniswapv3
My suggestion here is to keep both. 
Other solution would be to delete the uniswapv3 test and code as it is skipped

Comment (ClementWalter) [reply .gitmodules:18]: or make the submodule name explicit

Comment (obatirou) [reply .gitmodules:18]: Done 
Should be better

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**